sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -157,4 +161,38 @@ private[sql] class CacheManager extends Logging {
       case _ =>
     }
   }
+
+  /**
+   * Invalidates the cache of any data that contains `resourcePath` in one or more
+   * `HadoopFsRelation` node(s) as part of its logical plan.
+   */
+  private[sql] def invalidateCachedPath(
+      sparkSession: SparkSession, resourcePath: String): Unit = writeLock {
+    val (fs, qualifiedPath) = {
+      val path = new Path(resourcePath)
+      val fs = path.getFileSystem(sparkSession.sessionState.newHadoopConf())
+      (fs, path.makeQualified(fs.getUri, fs.getWorkingDirectory))
+    }
+    cachedData.foreach {
+      case data if data.plan.find {
```

Comment:
Could you move this into a separate function; it was kinda hard to understand that it is a part of the `case` guard.